tests_e2e/tests/lib/network_security_rule.py

```diff
@@ -76,8 +76,14 @@ def _get_network_security_group(self) -> Dict[str, Any]:
                 "securityRules": []
             }}
         }}""")
-        nsg_reference = "network_security_groups"
-        resources[nsg_reference] = network_security_group
+
+        # resources is a dictionary in LISA's ARM template, but a list in the template for scale sets
+        if isinstance(resources, dict):
```

Comment:
resources is typed as a dict on line 58. Should the type annotation be removed?
`resources: Dict[str, Dict[str, Any]] = self._template["resources"]`

Reply:
updated